platform/resource-bundle-locator/src/main/java/ddf/platform/resource/bundle/locator/ResourceBundleLocatorImpl.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.platform.resource.bundle.locator;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+
+public class ResourceBundleLocatorImpl implements ResourceBundleLocator {
+
+  private File resourceBundleBaseDir;
+
+  @Override
+  public ResourceBundle getBundle(String baseName) {
+    return getBundle(baseName, Locale.getDefault());
+  }
+
+  @Override
+  public ResourceBundle getBundle(String baseName, Locale locale) {
+    File resourceBundleDir = resourceBundleBaseDir;
+    if (resourceBundleDir == null) {
+      resourceBundleDir = new File(String.format("%s/etc/i18n/", System.getProperty("ddf.home")));
+    }
+
+    try {
+      URL[] urls = {resourceBundleDir.toURI().toURL()};
+
+      try (URLClassLoader loader = new URLClassLoader(urls)) {
+        return ResourceBundle.getBundle(baseName, locale, loader);
+      }
+
+    } catch (MalformedURLException e) {
+      throw new MissingResourceException(
+          "An error occurred while loading ResourceBundle: " + baseName + "," + locale.getCountry(),
+          getClass().getName(),
+          baseName);
+    } catch (IOException e) {
+      throw new MissingResourceException(
+          "An error occurred while creating class loader to URL for ResourceBundle: "
+              + baseName
+              + ","
+              + locale.getCountry(),
```

Comment:
❓ Why only `getCountry()`? Resource bundles are loaded based on a locale's components and tight together. So in fact for a en_US_xxx locale, it would have attempted to find the bundle with en_US_xxx, then en_US, then en and then the default one without locale. If all of them are not found then you get an error. If one or more are found, they are linked together before being returned.

Reply:
This is a very good point. This should make the fallback procedure more traceable.

Reply:
As I mentioned in another comment, at a previous company we would always provide a default resource bundle which was a catch all for all locales. That one had to be provided as a class and not as a resource file. This ensured that you could never get into the missing resource exception case since your code was designed with the fallback resource in place.